sources_test/string.cpp

```diff
@@ -0,0 +1,125 @@
+#include "fixture.hpp"
+#include "string/parser.hpp"
+#include "file_system/read_target.hpp"
+#include "file_system/file_read.hpp"
+
+namespace kustest
+{
+class UTestString : public Fixture {};
+
+TEST_F(UTestString, test_slice_empty_input)
+{
+    std::string_view s;
+    char buffer[256];
+    std::string delimiters = ", ";
+    std::string erase = "!";
+
+    auto result = str::Parser::parse_in_new(s, buffer, delimiters, erase);
+    ASSERT_EQ(result.size(), 0);
+}
+
+TEST_F(UTestString, test_slice_empty_output)
+{
+    std::string_view s = ",!,,, ,,!";
+    char buffer[256];
+    std::string delimiters = ", ";
+    std::string erase = "!";
+
+    auto result = str::Parser::parse_in_new(s, buffer, delimiters, erase);
+    ASSERT_EQ(result.size(), 0);
+}
+
+TEST_F(UTestString, test_slice_simple)
+{
+    std::string_view s = "Hello, world!";
+    char buffer[256];
+    std::string delimiters = ", ";
+    std::string erase = "!";
+
+    auto result = str::Parser::parse_in_new(s, buffer, delimiters, erase);
+    ASSERT_EQ(result.size(), 2);
+    ASSERT_EQ(result[0], "Hello");
+    ASSERT_EQ(result[1], "world");
+}
+
+TEST_F(UTestString, test_slice_repeated_delimiters) {
+    std::string_view s = "!Hello!, ,world!!,,  , , !!";
+    char buffer[256];
+    std::string delimiters = ", ";
+    std::string erase = "!";
+
+    auto result = str::Parser::parse_in_new(s, buffer, delimiters, erase);
+    ASSERT_EQ(result.size(), 2);
+    ASSERT_EQ(result[0], "Hello");
+    ASSERT_EQ(result[1], "world");
+}
+
+TEST_F(UTestString, test_slice_into_old_buffer)
+{
+    std::string s = "apple,,,orange...banana!grape!!pear--peach";
```

Comment:
`std::string s = "appleA Zorange.\t.banana!grape!!pear--peach";`
(or something like that, separated by one char, also A Z a z)